src/sudoers/ast.rs

```diff
@@ -316,13 +317,10 @@ impl Parse for UserSpecifier {
         let begin_pos = stream.get_pos();
         if stream.eat_char('"') {
             let Unquoted(text, _): Unquoted<Username> = expect_nonterminal(stream)?;
-            let result = parse_user(&mut CharStream::new(text.chars()));
-            if result.is_err() {
-                unrecoverable!(pos = begin_pos, stream, "invalid user")
-            }
+            let result = parse_user(&mut CharStream::new_with_pos(&text, advance(begin_pos, 1)))?;
```

Comment:
Calling `get_pos()` after `stream.eat_char()` would work too, right?

Reply:
That simplifies this part a bit, yes.